### DISCUSSION TITLE
Guide: How I debugged my board not showing up

### DISCUSSION THREAD

**Discussion by kxing28** [NUS-CG3207/labs#8] [Wrangling Vivado] [answered]
For some reason my board (Nexys 4) wasn't showing up in Vivado when I ran "Open Hardware Target". Here's my guide in case anyone else ran into the issue.

1. I went to windows "Device Manager" and verified "USB Serial Converter A" and "USB Serial Converter B" appears when the device is plugged in.
![image](https://github.com/user-attachments/assets/3875eba8-e512-44f0-91fd-8433fd10e971)
2. I right-clicked on each one and went Properties->Advanced and I made sure "Load VCP" was unchecked. After reconnecting the device, it showed up in "Open Hardware Target".

If those Serial Converter lines aren't showing up in Device Manager when connecting the board, then you need to reinstall some drivers.
3. From Vivado, I clicked on the top bar Help->Add Design Tools or Devices. This launches the Vivado installer again, we just want to make sure we install the cable drivers again. Make sure "Install Cable Drivers" is checked.
![image](https://github.com/user-attachments/assets/0883ea76-bfab-464a-9c0d-abe269d892ef)

If your device still doesn't show up in "Open Hardware Target", you'll need to install "Adept.
4. I installed "Adept Runtime" and "Adept System" from https://digilent.com/reference/software/adept/start?redirect=2#software_downloads . Make sure the board is unplugged during this installation.
5. I opened "Adept" and plugged in my device, and verified that it was recognized in Adept. You'll need to close your "Open Hardware Target" window during this. Now go back to steps 1. and 2. to remove the extra COM Port.

At the end of this, I was able to find it in "Open Hardware Target". I guess most students here already have this configured, but maybe this will be helpful to other exchangers.

**Comment by NeilBaner** [comment, marked as answer]
Thanks very much for sharing! We will include these steps in the Vivado installation guide, for greater visibility, with credit to you of course.

**Reply by kxing28** [reply]
Oh that's cool, thanks!
